File: utils/cache.py
import cachetools

class CacheManager:
    def __init__(self):
        self.local_cache = cachetools.LRUCache(maxsize=1000)

    def fetch_from_cache(self, key):
        return self.local_cache.get(key)

    def store_in_cache(self, key, value):
        self.local_cache[key] = value

File: utils/cache.ts
import { LRUCache } from 'cachetools';

class CacheManager {
    private localCache: LRUCache<string, any>;

    constructor() {
        this.localCache = new LRUCache({ maxSize: 1000 });
    }

    fetchFromCache(key: string): any {
        return this.localCache.get(key);
    }

    storeInCache(key: string, value: any): void {
        this.localCache.set(key, value);
    }
}

export default CacheManager;
